File: LeetCodeProblemsCMake/Problems/LeetCode2154KeepMultiplyingFoundValuesByTwo.cpp
//
// Created by saeed on 11/19/25.
//

#include "LeetCode2154KeepMultiplyingFoundValuesByTwo.h"


#include <unordered_set>
using namespace std;

int LeetCode2154KeepMultiplyingFoundValuesByTwo::findFinalValue(vector<int> &nums, int original) {
    unordered_set<int> numsSet{};
    for (auto num: nums) {
        numsSet.insert(num);
    }
    while (numsSet.contains(original)) {
         original<<=1;
    }
    return original;
}

File: LeetCodeProblemsGo/LeetCode2154KeepMultiplyingFoundValuesByTwo/LeetCode2154KeepMultiplyingFoundValuesByTwo.go
package LeetCode2154KeepMultiplyingFoundValuesByTwo

func findFinalValue(nums []int, original int) int {
	numsMap := make(map[int]any)
	for _, num := range nums {
		numsMap[num] = struct{}{}
	}
	for true {
		if _, exists := numsMap[original]; !exists {
			return original
		}
		original <<= 1
	}
	return original
}
